CRM/Odoosync/Form/Configurations.php

```diff
@@ -0,0 +1,212 @@
+ */
+class CRM_Odoosync_Form_Configurations extends CRM_Core_Form {
+
+  /**
+   * Contains array of fields, which config Odoo Civicrm Sync Expansion
+   *
+   * @var string[]
+   */
+  private $settingFields = [];
+
+  /**
+   * Builds the form object.
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  public function buildQuickForm() {
+    $configElements = [];
+    CRM_Utils_System::setTitle(E::ts('CiviCRM Odoo Sync Configuration'));
+
+    $settingFields  = $this->getSettingFields();
+
+    foreach ($settingFields as $name => $config) {
+      $this->add(
+        $config['html_type'],
+        $name,
+        E::ts($config['title']),
+        CRM_Utils_Array::value('html_attributes', $config, []),
+        $config['is_required'],
+        CRM_Utils_Array::value('extra_data', $config, [])
+      );
+
+      $configElements[] = $name;
+    }
+
+    $this->addButtons([
+      [
+        'type' => 'submit',
+        'name' => E::ts('Submit'),
+        'isDefault' => TRUE,
+      ],
+      [
+        'type' => 'cancel',
+        'name' => E::ts('Cancel'),
+      ],
+    ]);
+
+    $this->assign('configElements', $configElements);;
+  }
+
+  /**
+   * Process the form after the input has been submitted and validated.
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  public function postProcess() {
+    $this->controller->setDestination(
+      CRM_Utils_System::url(
+        'civicrm/admin/odoosync/configuration',
+        http_build_query(['reset' => 1])
+      )
+    );
+
+    $settingFields = $this->getSettingFields();
+    $submittedValues = $this->exportValues();
+    $valuesToSave = array_intersect_key($submittedValues, $settingFields);
+
+    //if empty submitted value, set default data
+    foreach ($settingFields as $key => $val) {
+      $isEmptySubmittedValue = !array_key_exists($val['name'], $valuesToSave) || empty($valuesToSave[$val['name']]);
+      if ($isEmptySubmittedValue && !empty($val['default'])) {
+        $valuesToSave[$val['name']] = $val['default'];
+      }
+    }
+
+    civicrm_api3('setting', 'create', $valuesToSave);
+  }
+
+  /**
+   * Gets the configurations allowed to be set on this form.
+   *
+   * @return array
+   * @throws \CiviCRM_API3_Exception
+   */
+  private function getSettingFields() {
+    if (!empty($this->settingFields)) {
+      return $this->settingFields;
+    }
+
+    $this->settingFields = self::getSettingValues();
+
+    return $this->settingFields;
+  }
+
+  /**
+   * Gets setting values used by the extension
+   *
+   * @return mixed
+   * @throws \CiviCRM_API3_Exception
+   */
+  public static function getSettingValues() {
+    return civicrm_api3('setting', 'getfields', [
+      'filters' => [ 'group' => 'odoosync'],
+    ])['values'];
+  }
+
+  /**
+   * Sets defaults for form.
+   *
+   * @see CRM_Core_Form::setDefaultValues()
+   * @throws \CiviCRM_API3_Exception
+   */
+  public function setDefaultValues() {
```

Comment:
it more logical and easier to read to move this method before postProcess, since it happen before the form post processing.